src/xrdsvc/StreamBuffer.cc

```diff
@@ -154,6 +154,10 @@ void StreamBuffer::cancel() {
     // better a possible leak than corrupted memory or a permanently wedged
     // thread in a limited pool.
     // In any case, this code having an effect should be extremely rare.
+    // FUTURE: It would be nice to eliminate this possible memory leak.
+    //       Possible fix, atomic<bool> _recycleCalled, create thread
+    //       to check if _recycleCalled == true. If true or 24 hours pass
+    //       use `Ptr keepAlive = std::move(_selfKeepAlive);` to kill the object.
     {
```

Comment:
Is this really the "leak" or an extended lifespan of an object? If this is the former then it has to be investigated and fixed right away. If this is the latter then the proposed time-based cleanup may be an option before we have an extended Czar-to-workers communication protocol for sending direct query cancellation/completion signals to workers.

Reply:
There's an ugly race condition where it's difficult to tell if xrootd is going to call Recycle() or not. I've been thinking about it. Waiting a very long time would work, but it's ugly. I'm also thinking that when the SsiRequest is getting deleted, it's probably safe to say that xrootd isn't going to be calling Recycle(), but I'm not 100% sure about that.

Reply:
I have filed the JIRA ticket for this race condition: https://jira.lsstcorp.org/browse/DM-39041